ADR/0047-creating-pending-pipelineruns.md

```diff
@@ -0,0 +1,71 @@
+# 47. Creating pending pipeline runs
+
+Date: 2024-11-21
+
+## Status
+
+Accepted
+
+## Context
+
+When pipelines start, the time is immediately being taken away from the configured timeout (even if their pods are waiting for resources). While this behavior is likely sufficient when fewer concurrent pipelines are running, timeouts are more likely to be exceeded when bulk builds are triggered (i.e. a Renovate job updating common dependencies). While the speed at which a pipeline completes is important, a delayed and completed pipeline will likely finish quicker than an immediately and timeout-killed pipeline. Futhermore, having multiple pipelines running at the same time may starve other concurrent pipelines from acquiring the necessary resources resulting in fewer completed pipelines.
+
+So say you started 300 pipeline runs on Friday evening and you expect to see 300 new images by Monday morning.
+You have resource quota for 20 concurrent builds, so 20 builds get started, when they finish 20 more and so on.
+But because all 300 pipeline runs started (from Tekton point of view) when they were created, after few iterations all the remaining pipeline runs will be cancelled (due to a timeout).
+
+When many pipelines are triggered across multiple namespaces, there is also an increased load on etcd which may degrade the cluster's overall performance.
+
+## Decision
+
+We will use idea of creating pipeline runs in pending state and having a standalone controller that would implement any custom logic on when to start each pending pipeline:
```

Comment:
@jhutar , do you want to update this to use https://kueue.sigs.k8s.io/docs/overview/ instead of a custom controller?

Reply:
I talked with @gbenhaim today briefly. A controller may still be used even if integrating with kueue.